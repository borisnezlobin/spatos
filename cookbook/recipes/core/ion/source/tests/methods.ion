let array = [ one two three four ]
let space_string = $join(array)
let comma_string = $join(array ', ')
echo $space_string
echo $comma_string
echo @split(space_string)
echo @split(comma_string ', ')

let array = ["one two" "three four" "five six" "seven eight" "nine ten"]
echo $len(@array)
for element in 0..$len(@array)
    echo @array[$element]
end

let string = "one ðŸ˜‰ðŸ˜‰ðŸ˜‰ two ðŸ˜‰ðŸ˜‰ðŸ˜‰ three ðŸ˜‰ðŸ˜‰ðŸ˜‰ four ðŸ˜‰ðŸ˜‰ðŸ˜‰ five"
echo $len(string) $len_bytes(string)
for grapheme in 0..$len(string)
    echo $string[$grapheme]
end

echo $replace("one two one two" one 1)
echo $replacen("one one one one" one 1 3)
echo $repeat("one " 5)

echo $join([one two three] "\n")
echo $join([one two three] "\t")
echo $join([one two three] '\\n')
echo $join([one two three] "\\\\n")
echo $replace($join([one two three] "\n") "\n" "\t")
let a = "applesauce"
let pos = $find(a "s")
let array = [@split_at(a $pos)]
echo $join(array "\n")

let a = [1 2 3 4 5]
let b = "1 2 3 4 5"
echo $join(@split(b " ") $join(a " "))

echo $regex_replace("one two onemy anemy town" "\ o|\ a" "\ e")

echo $unescape('one two\"\tone')
echo $escape("'one two'")

let spacey = "  Spacey "
echo $trim(" So Space  ")!
echo $trim($spacey)!
echo $trim_end(" So Space  ")!
echo $trim_end($spacey)!
echo $trim_start(" So Space  ")!
echo $trim_start($spacey)!
