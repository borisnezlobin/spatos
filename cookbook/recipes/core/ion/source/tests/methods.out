one two three four
one, two, three, four
one two three four
one two three four
5
one two
three four
five six
seven eight
nine ten
39 75
o
n
e
 
ğŸ˜‰
ğŸ˜‰
ğŸ˜‰
 
t
w
o
 
ğŸ˜‰
ğŸ˜‰
ğŸ˜‰
 
t
h
r
e
e
 
ğŸ˜‰
ğŸ˜‰
ğŸ˜‰
 
f
o
u
r
 
ğŸ˜‰
ğŸ˜‰
ğŸ˜‰
 
f
i
v
e
1 two 1 two
1 1 1 one
one one one one one 
one
two
three
one	two	three
one\ntwo\nthree
one\ntwo\nthree
one	two	three
apple
sauce
11 2 3 4 521 2 3 4 531 2 3 4 541 2 3 4 55
one two enemy enemy town
one two"	one
\'one two\'
So Space!
Spacey!
 So Space!
  Spacey!
So Space  !
Spacey !
